new_arrivals_chi/app/main.py

```diff
@@ -517,6 +530,97 @@ def edit_organization():
     )
 
 
+@main.route("/add_organization_success", methods=["GET"])
+@login_required
+def add_organization_success():
+    """Establishes route to the add organization success page.
+
+    This route is accessible by selecting 'Dashboard' on the
+    home page.
+
+    Returns:
+        Renders the add organization success page where admin can view
+        a success message after adding a new organization.
+    """
+    language = bleach.clean(request.args.get("lang", "en"))
+    translations = current_app.config["TRANSLATIONS"][language]
+    return render_template(
+        "add_organization_success.html",
+        language=language,
+        translations=translations,
+    )
+
+
+@main.route("/add_organization", methods=["GET", "POST"])
+@login_required
+def add_organization():
+    """Establishes route to the add organization page.
+
+    This route is accessible by selecting 'Dashboard' on the
+    home page.
+
+    Returns:
+        Renders the add organization page where admin can add
+        a new organization.
+    """
+    # Check if the user is an admin
+    if current_user.role != "admin":
+        return "Unauthorized", 401
+
+    language = bleach.clean(request.args.get("lang", "en"))
+    translations = current_app.config["TRANSLATIONS"][language]
+
+    if request.method == "POST":
+        email = request.form.get("email")
+        confirmed_email = request.form.get("email-confirm")
+        phone_number = request.form.get("phone-number")
+        org_name = request.form.get("org-name")
+
+        # Check if the email and confirmed email match
+        if email != confirmed_email:
+            flash(escape("Emails do not match. Try again"))
+        # Handle the form submission
+        elif not validate_email_syntax(email):
+            flash(escape("Invalid email address. Try again"))
+
+        elif not validate_phone_number(phone_number):
+            flash(
+                escape("Invalid phone number (correct example: ###-###-####)")
+            )
+        else:
+            # Create the organization as HIDDEN
+            new_org_id = create_organization_profile(
+                org_name, phone_number, "HIDDEN"
+            )
+
+            # Create a temporary password for the user
+            temp_pwd = create_temp_pwd(email, phone_number)
+
+            # Create the user with the provided email and a default password
+            new_user = create_user(email, temp_pwd)
+
+            # Update the user with the new organization
+            user = User.query.get(new_user.id)
+            if user:
+                user.organization_id = new_org_id
+                db.session.commit()
+            else:
+                raise Exception("User not found")
```

Comment:
we probably want to do a try catch here so that the exception doesnt crash the app

Reply:
Done!